File: c/practice/q2.m
length = 10;
guanine = 0;
adenine = 0;
cytosine = 0;
thymine = 0;


dnaSequence = [length];
for i = 1 : length
    dnaSequence(i) = input("Please enter the DNA Sequence: ",'s');
    while (~(dnaSequence(i) == 'G' || dnaSequence(i) == 'T' || ...
            dnaSequence(i) == 'A' || dnaSequence(i) == 'C'))
        dnaSequence(i) = input("Wrong input enter again: ",'s');
    end
    if (dnaSequence(i) == 'G')
        guanine = guanine + 1;
        
    elseif(dnaSequence(i) == 'T')
        adenine = adenine + 1;
        
    elseif( dnaSequence(i) == 'A' )
        cytosine = cytosine + 1;
        
    else
        thymine = thymine + 1;
    end
end

fprintf("Sequence\n");
for i = 1 : length
    fprintf("%c", dnaSequence(i));
end

fprintf("\n");
fprintf("%d guanine\n", guanine);
fprintf("%d adenine\n",adenine);
fprintf("%d cytosine\n", cytosine);
fprintf("%d thymine\n", thymine);

File: c/practice/q2.c
# include<stdio.h>

int main(void){
    int length = 10;
    char dnaSequence[length];
    int guanine = 0;
    int adenine = 0;
    int cytosine = 0;
    int thymine = 0;
    for(int index = 0; index < length; index++){
        printf("Please enter the input: ");
        scanf("%c%*c", &dnaSequence[index]);
        while(dnaSequence[index] != 'G' && dnaSequence[index] != 'A' && dnaSequence[index] != 'T' && dnaSequence[index] != 'C' ){
            printf("Please enter a correct input: ");
            scanf("%c%*c", &dnaSequence[index]);
        }
        if(dnaSequence[index] == 'G'){
            guanine++;
        }
        else if(dnaSequence[index] == 'A'){
            adenine++;
        }
        else if(dnaSequence[index] == 'T'){
            thymine++;
        }
        else{
            cytosine++;
        }
    }
    printf("Sequence \n");
    for (int index=0; index < length; index++) {
        printf("%c", dnaSequence[index]);
    }
    printf("\n");
    printf("%d guanine\n", guanine);
    printf("%d adenine\n",adenine);
    printf("%d cytosine\n", cytosine);
    printf("%d thymine\n", thymine);
}

